client/web/src/enterprise/code-monitoring/CreateCodeMonitorPage.tsx

```diff
@@ -227,33 +294,41 @@ const TriggerArea: React.FunctionComponent<TriggerAreaProps> = ({
                         <div className="create-monitor-page__query-input">
                             <input
                                 type="text"
-                                className="form-control my-2"
-                                onChange={event => {
-                                    onQueryChange(event.target.value)
-                                }}
-                                value={query}
+                                className="create-monitor-page__query-input-field form-control my-2"
```

Comment:
```suggestion
                                className={classnames(
                                    'create-monitor-page__query-input-field form-control my-2',
                                    deriveInputClassName(queryState)
                                )}
```

`deriveInputClassName` comes from `useInputValidation.ts`.

We're using our `<Form />` component, which handles `InvalidEvent`s on submission and applies validation styling to the whole form. We want to handle this input's styling on change, not on submission, so set classes based on validation state.

Reply:
Ah got it, thanks!